docs/manual/cli.qmd

```diff
@@ -0,0 +1,151 @@
+```{python}
+#| echo: false
+from odm_sharing.tools.share import OutFmt, share
+
+from common import DATA, SCHEMA, load_csv_md, print_file
+```
+
+# CLI {#sec-cli}
+
+## Reference
+
+```bash
+odm-share [OPTION]... SCHEMA INPUT
+```
+
+Arguments:
+
+- SCHEMA
+
+  sharing schema file path
+
+- INPUT
+
+  spreadsheet file path or [SQLAlchemy database URL](https://docs.sqlalchemy.org/en/20/core/engines.html#database-urls)
+
+Options:
+
+- `--orgs=NAME[,...]`
+
+    comma separated list of organizations to output data for, defaults to all
+
+- `--outfmt=FORMAT`
+
+    output format (excel or csv), defaults to input format
+
+- `--outdir=PATH`
+
+    output file directory, defaults to the current directory
+
+- `-d`, `--debug`:
+
+    output debug info to STDOUT (and ./debug.txt) instead of creating sharable
+    output files. This shows which tables and columns are selected, and how
+    many rows each filter returns.
+
+- `-q`, `--quiet`:
+
+    don't log to STDOUT
+
+One or multiple sharable output files will be created in the chosen output
+directory according to the chosen output format and organization(s). Each
+output file will have the input name followed by the recipient org name.
+
+### Errors
+
+Error messages will be printed to the terminal (STDERR) when something is
+wrong. The message starts with telling where the error originated, including
+the filename and line number or rule id. Here's a few examples:
+
+When headers are missing from the schema:
+
+```{python}
+#| echo: false
+share('assets/schema-missing-headers.csv', 'assets/measures.csv')
+```
+
+When no share-rules are contained in the schema:
+
+```{python}
+#| echo: false
+share('assets/schema-missing-rules.csv', 'assets/measures.csv')
+```
+
+## Usage
+
+### Examples
+
+#### CSV
+
+To share a single table `measures.csv`, using the sharing schema `schema.csv`,
+the following command can be used:
+
+```bash
+odm-share schema.csv measures.csv
+```
+
+It will make an output file called `measures-<org>.csv` for each organization
+specified in the schema, with filtered data that is ready to share.
+
+#### Excel
+
+Excel files can be used as input to share multiple tables at once:
+
+```bash
+odm-share schema.csv data.xlsx
+```
+
+It will make an output file called `<org>.xlsx` for each organization in the
+schema.
+
+#### Database
+
+To use a MySQL database as input (with the pymysql package):
+
+```bash
+odm-share schema.csv mysql+pymysql://user:pass@host/db
+```
+
+Same as above, using a MS SQL Server database through ODBC (with the pyodbc
+package):
+
+```bash
+odm-share schema.csv mssql+pyodbc://user:pass@mydsn
+```
+
+#### Options
+
+- Share CSV files from an Excel file:
```

Comment:
Don't think these need to be nested within another sub heading. They're just another example.

Reply:
the other sections on the same level as "options" are specific to csv, excel or database sources, while this "options" section is common to all sources. How would you change it?

Reply:
Sorry, I must have misread this title.